packages/cli/src/actions/PullAction.ts

```diff
@@ -0,0 +1,112 @@
+import { ExperimentApiService } from "../services/experiment/ExperimentApiService";
+import { ExperimentConfigModel } from "../config/ExperimentsConfig";
+import { ExperimentFlagComparator } from "../services/experiment/ExperimentFlagComparator";
+import { ComparisonResultSymbol, ICON_RETURN_ARROW, ICON_SUCCESS, ICON_WARNING_W_TEXT } from "../ui/icons";
+import { ComparisonResult } from "../comparison/ComparisonResult";
+import { cloneDeep } from "lodash";
+import { loadLocalConfiguration, saveYamlToFile } from "../config/AmplitudeConfigYamlParser";
+import { BaseAction } from "./BaseAction";
+
+export interface PullActionOptions {
+  config: string;
+  experimentManagementApiKey?: string;
+  experimentDeploymentId?: string;
```

Comment:
Deployment ID is impossible for customers to get without using the mgmt api themselves.

I would switch this to experimentDeploymentKey, then get the deployments, and find the object with the matching key and take its id.